src/software/full_system_main.cpp

```diff
@@ -56,53 +62,50 @@ void setBackendFromString(std::string backend_name)
  *
  * @return True if the program should continue, false otherwise
  */
-bool parseCommandLineArgs(int argc, char **argv)
+commandLineArgs parseCommandLineArgs(int argc, char **argv)
 {
+    commandLineArgs args;
     // Build one string with all the backend_names
-    std::vector<std::string> backend_names = BackendFactory().getRegisteredBackendNames();
+    std::vector<std::string> backend_names = BackendFactory::getRegisteredBackendNames();
     std::string all_backend_names =
         std::accumulate(std::begin(backend_names), std::end(backend_names), std::string(),
                         [](std::string &ss, std::string &s) { return ss + s + ", "; });
     std::string backend_help_str =
         "The backend that you would like to use, one of: " + all_backend_names;
 
+    options_description desc{"Options"};
+    desc.add_options()("help,h", boost::program_options::bool_switch(&args.help),
+                       "Help screen")(
+        "backend", value<std::string>(&args.backend_name)->default_value(""),
+        backend_help_str.c_str())("headless",
+                                  boost::program_options::bool_switch(&args.headless),
+                                  "Run without the Visualizer");
+
+    variables_map vm;
     try
     {
-        options_description desc{"Options"};
-        desc.add_options()("help,h", "Help screen")(
-            "backend", value<std::string>()->notifier(setBackendFromString)->required(),
-            backend_help_str.c_str())("headless",
-                                      boost::program_options::bool_switch(&headless),
-                                      "Run without the Visualizer");
-
-        variables_map vm;
         store(parse_command_line(argc, argv, desc), vm);
+        notify(vm);
 
-        // We only process notifications if "help" was not given, which allows us to
-        // avoid issues where required arguments are not required if "help" is given
-        if (vm.count("help"))
+        if (args.help)
         {
             std::cout << desc << std::endl;
-            return false;
-        }
-        else
-        {
-            notify(vm);
         }
     }
     catch (const error &ex)
     {
         std::cerr << ex.what() << '\n';
-        return false;
+        args.err = true;
+        return args;
     }
 
-    return true;
+    return args;
 }
 
 /**
  * Connects all the observers together
  */
-void connectObservers()
+void connectObservers(bool headless)
```

Comment:
Perhaps we could add to the doc comment to give a quick description of what the headless param is for.

Reply:
Headless param is for whether or not we run the Visualizer. if `headless=true` the Visualizer is not run. If `headless=false`, the Visualizer is run.